### PR TITLE
opensnitch: Fix build with Go > 1.17

### PR DESCRIPTION
**Pull request by onny**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by onny** [comment]
@ofborg eval

File: pkgs/tools/networking/opensnitch/daemon.nix [hunk 1]
```diff
@@ -69,8 +72,7 @@ buildGoModule rec {
     description = "An application firewall";
     homepage = "https://github.com/evilsocket/opensnitch/wiki";
     license = licenses.gpl3Only;
-    maintainers = [ maintainers.raboof ];
+    maintainers = with maintainers; [ raboof onny ];
```
Comment (raboof) [inline pkgs/tools/networking/opensnitch/daemon.nix:75]: :heart: thanks a ton!

File: pkgs/tools/networking/opensnitch/daemon.nix [hunk 2]
```diff
@@ -40,9 +40,12 @@ buildGoModule rec {
 
   nativeBuildInputs = [ pkg-config protobuf go-protobuf makeWrapper protoc-gen-go-grpc ];
 
-  vendorSha256 = null; #vendorSha256 = "";
+  vendorSha256 = "sha256-jWP0oF+jZRFMi5Y2y0SARMoP8wTKVZ8UWra9JNzdSOw=";
 
   preBuild = ''
+    # Fix inconsistent vendoring build error
```
Comment (raboof) [inline pkgs/tools/networking/opensnitch/daemon.nix:46]: Do we have an upstream issue reference for this?

File: pkgs/tools/networking/opensnitch/go.mod [hunk 3]
```diff
@@ -0,0 +1,16 @@
+module github.com/evilsocket/opensnitch/daemon
+
+go 1.14
```
Comment (raboof) [inline pkgs/tools/networking/opensnitch/go.mod:3]: should this be 1.14?

**Review by raboof — APPROVED** [review]

File: pkgs/tools/networking/opensnitch/go.mod (hunk 3 above)
Comment (onny) [reply pkgs/tools/networking/opensnitch/go.mod:3]: This is the upstream reference, I don't want to change it as long as it compiles with the latest go version (1.19)

File: pkgs/tools/networking/opensnitch/daemon.nix (hunk 2 above)
Comment (onny) [reply pkgs/tools/networking/opensnitch/daemon.nix:46]: Okay added an upstream bug report

**Comment by onny** [comment]
Upstream bug report https://github.com/evilsocket/opensnitch/issues/770

**Comment by onny** [comment]
Result of `nixpkgs-review pr 201947` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>opensnitch</li>
  </ul>
</details>

**Comment by onny** [comment]
@ofborg eval

**Comment by onny** [comment]
Result of `nixpkgs-review pr 201947` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>opensnitch</li>
  </ul>
</details>

**Pull request merged**